torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -372,232 +372,270 @@ void SetExportModuleMobileInfoConverter(
-    class_deps_.add(root_type);
-    for (size_t i = 0; i < class_deps_.size(); ++i) {
-      // note: convertNameType may extend class_deps_, so re-checking
-      // .size() is necessary
-      convertNamedType(class_deps_[i]);
-    }
+void ScriptModuleSerializer::writeByteCode(
+    const Module& module,
+    bool save_mobile_debug_info) {
+  std::vector<c10::IValue> elements;
+  elements.emplace_back(
+      static_cast<int64_t>(caffe2::serialize::kProducedBytecodeVersion));
+  c10::optional<std::vector<c10::IValue>> debug_info_elements;
+  if (save_mobile_debug_info) {
+    debug_info_elements = std::vector<c10::IValue>();
+    debug_info_elements->emplace_back(
+        static_cast<int64_t>(caffe2::serialize::kProducedBytecodeVersion));
+  }
 
-    // Mapping of filename => src. We need this because multiple classes may go
-    // in the same file (e.g. foo.bar.Baz and foo.bar.Qux)
-    for (auto& item : file_streams_) {
-      const std::string filename = qualifierToArchivePath(item.key(), "code/");
-
-      std::string src = item.value().str();
-
-      // Only compress these records if they're not tiny.
-      // The cpu cost of generating zip datastructs and compressing isn't
-      // well-spent for very small records.
-      static constexpr size_t kMinToCompress = 200;
-
-      writer_.writeRecord(
-          filename,
-          src.c_str(),
-          src.size(),
-          src.size() > kMinToCompress /*compress*/);
-
-      // Write out the debug information
-      std::string debugFilename = filename + ".debug_pkl";
-      SourceRangePickler source_range_pickler;
-      auto range_data = source_range_pickler.pickle(item.value().ranges());
-      writer_.writeRecord(
-          debugFilename,
-          range_data.data(),
-          range_data.size(),
-          range_data.size() > kMinToCompress /*compress*/);
-    }
+  moduleMethodsTuple(
+      module, elements, debug_info_elements, save_mobile_debug_info);
+  auto telements = Tup(std::move(elements));
+  uint64_t next_tensor_id =
+      writeArchive(telements, "bytecode", "", "bytecode/", 0);
+  if (save_mobile_debug_info) {
+    auto debug_info_telements = Tup(std::move(debug_info_elements.value()));
+    writeArchive(
+        debug_info_telements,
+        "mobile_debug",
+        "",
+        "mobile_debug/",
+        next_tensor_id);
   }
+}
 
-  void writeByteCode(const Module& module, bool save_mobile_debug_info) {
-    std::vector<c10::IValue> elements;
-    elements.emplace_back(
-        static_cast<int64_t>(caffe2::serialize::kProducedBytecodeVersion));
-    c10::optional<std::vector<c10::IValue>> debug_info_elements;
-    if (save_mobile_debug_info) {
-      debug_info_elements = std::vector<c10::IValue>();
-      debug_info_elements->emplace_back(
-          static_cast<int64_t>(caffe2::serialize::kProducedBytecodeVersion));
-    }
+std::tuple<uint64_t, uint64_t> ScriptModuleSerializer::serialize_unified_format(
```

Comment:
This method is the only read 'new' logic in this PR, the rest of it is refactoring of the already existing logic